pkg/lock/simple_mem.go

```diff
@@ -0,0 +1,30 @@
+package lock
+
+import "sync"
+
+var locks map[string]*memLock
```

Comment:
What is the purpose of the `memlock` struct? `var locks map[string]sync.Mutex` is not enough?